src/store/validation/validationHelpers.ts

```diff
@@ -0,0 +1,127 @@
+  if (subSchema === '') {
+    return ajv.compile(jsonSchema)
+  }
+  let validate = ajv.getSchema(subSchema)
+  if (!validate) {
+    ajv.addSchema(jsonSchema)
+    validate = ajv.getSchema(subSchema)
+  }
+  if (!validate) {
+    throw new Error(`Cannot find definition ${subSchema} in schema ${jsonSchema}`)
+  }
+  return validate as ValidateFunction
+}
+
+export function debugValidationStatus(status: IValidationStatus, customMessage: string): void {
+  if (status != null) {
+    console.debug(
+      'store/validation:',
+      customMessage,
+      ' | isValid:',
+      status.isValid,
+      ' | error dataPaths:',
+      `${Object.prototype.hasOwnProperty.call(status, 'ajvErrors') && status.ajvErrors ? (status.ajvErrors).map((item) => get(
+        item,
+        'dataPath',
+        'undefined',
+      )).join(';') : 'undefined'}`,
+      ' | error messages:',
+      `${Object.prototype.hasOwnProperty.call(status, 'ajvErrors') && status.ajvErrors ? (status.ajvErrors).map((item) => get(
+        item,
+        'message',
+        'undefined',
+      )).join(';') : 'undefined'}`,
+      ' | error details:',
+      status,
+    )
+  } else {
+    console.debug('store/validation:', 'the status in debugValidationStatus was undefined')
+  }
+}
+
+export function flatValidationStatuses({
+  keyPrefix,
+  errors,
+  accumulator,
+}: { keyPrefix: string, errors: undefined | null | Array<ErrorObject>, accumulator: IIndexedString }): void {
+  errors?.forEach((error) => {
+    let index = ''
+    let message = ''
+    if (DEV_ERROR_KEYWORDS.includes(error.keyword)) {
+      // this is more likely a dev issue than user error
+      // error.dataPath could be empty or not for such errors
+      index = `${keyPrefix}${error.schemaPath}`
+      message = `${error.message}, for params ${JSON.stringify(error.params)}`
+      console.warn('store/validation:', `Schema issue found on ${index}: ${message}`)
+    } else {
+      index = `${keyPrefix}${error.dataPath}`
+      message = error.message as string
+    }
+    accumulator[index] = message
+  })
+}
+
+export function getOrCreateFlowValidator(schemaVersion: string): ValidateFunction {
+  const validationType = 'flow'
+  if (isEmpty(validators) || !validators.has(validationType)) {
+    const flowJsonSchema = require(`@floip/flow-runner/dist/resources/validationSchema/${schemaVersion}/flowSpecJsonSchema.json`)
+
+    // remove `blocks` property from IFlow schema to avoid double validations
+    flowJsonSchema.definitions.IFlow.additionalProperties = true
+    delete flowJsonSchema.definitions.IFlow.properties.blocks
+
+    validators.set(validationType, createDefaultJsonSchemaValidatorFactoryFor(flowJsonSchema, '#/definitions/IFlow'))
+  }
+  return validators.get(validationType)!
+}
+
+export function validate_community_block({block, schemaVersion}: {block: IBlock, schemaVersion: string}): IValidationStatus {
```

Comment:
Let's keep method name cameCase in this helpers: `validateCommunityBlock`

Normally, even for methods under `validation` module, we could avoid snake_casing there (no need to updated them now for `validate_flow` etc, I'm just explaining).
To my understanding, we have snake_cased prefix (eg: `block_setName`, `resource_delete`, etc) if we have multiple vuex module (block, resources, flow, etc) under 01 name space (`flow`)
<img width="166" alt="Screen Shot 2021-08-24 at 6 29 41 PM" src="https://user-images.githubusercontent.com/68745918/130707635-f20e60e9-b04d-455e-869c-0d2bb8a4cb64.png">

That helps us to have better code readability, and gives us a clue where to see the method implementation if we have something like:
- dispatch(`flow/validate_flow`, ...)
- commit(`flow/resource_delete`, ...)

Reply:
And I suggest we use camelCase for `validateBlock` in each specific block type module as well.

Reply:
I agree, done!